platform-plugins.md

```diff
@@ -0,0 +1,277 @@
+generation, but rather on a flexible message passing style:
+
+* The Flutter portion of your app sends messages to it's *host*, the iOS or
+ Android portion of your app, over a platform channel.
+
+* The *host* listens on the platform channel, and receives the message. It then
+ calls into any number of platform-specific APIs -- using their native
+ programming language -- and sends back a response to the *client*, the Flutter
+ portion.
+
+### Architectural overview: Platform Channels
+
+Messages are passed between the client (UI) and host (platform) using Platform
+channels as illustrated in this diagram:
+
+![Platform Channels architecture](/images/PlatformChannels.png)
+
+Message and responses are passed asynchronously, to ensure the user interface
+remains responsive.
+
+On the client side, the class
+[`PlatformMessageChannel`](https://docs.flutter.io/flutter/services/PlatformMessageChannel-class.html)
+enables sending general purpose message, and the class
+[`PlatformMethodChannel`](https://docs.flutter.io/flutter/services/PlatformMethodChannel-class.html)
+enables spending specific messages that correspond to method calls. On the
+platform side, the class `FlutterMessageChannel` enables receiving messages
+messages from the client, and responding with a result, and the class
+`FlutterMethodChannel`. These classes allow you to developer a platform plugin
+with very little 'boilerplate' code.
+
+### Example: Calling platform-specific iOS and Android code using Platform Channels
+
+The following example demonstrates how to call a new platform-specific API,
+through a concrete example that retrieves and displays the current battery level
+by calling into the Android `BatteryManager` API, and the iOS
+`device.batteryLevel` API, via a single platform message, `getBatteryLevel`.
+
+The standard app template created when you create a new Flutter app can be used
+
+*Note*: The full source-code for this example is available in
+[`/samples/platform-services/`](https://github.com/flutter/flutter/tree/master/examples/platform_services).
+
+**Step 1: Create the Flutter platform client**
+
+The Flutter platform client code is written inside the app's `State` class.
+Start by constructing the channel. Since we need just a single platform message,
+`getBatteryLevel`, representing a platform method call, we will use a
+`PlatformMethodChannel`:
+
+```dart
+class _PlatformServicesState extends State<PlatformServices> {
+  static const PlatformMethodChannel platform = const PlatformMethodChannel('battery');
+  ...
+}
+```
+
+The client and host sides of the channel are connected through by the passed
+name argument; here `battery`. Note that channel names must be unique; we
+recommend using a naming pattern using *TODO*.
+
+Next, we invoke a method on the method channel, specifying the concrete method
+to call via a String `method` argument, and use the returned result to update
+our user interface:
+
+```dart
+String _batteryLevel = 'Unknown battery level.';
```

Comment:
One of the pieces of feedback we got from the previous event was "I lost my place as I was adding the code from the doc into my project". I think a simple comment like "// inside of the foo() method" would be sufficient for snippets like this. A little "sign post".